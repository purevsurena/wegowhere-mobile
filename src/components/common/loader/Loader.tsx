import * as React from 'react';
import { Colors } from '@/constants';
import { Flex } from '@/components/layouts';
import { mScale, scale } from '@/utils/Metrics';
import {
  StyleSheet,
  ActivityIndicator,
  Dimensions,
  Platform,
} from 'react-native';

const Loader = () => {
  return (
    <Flex style={styles.container}>
      <Flex jc="center" ai="center" style={styles.loaderContainer}>
        <ActivityIndicator
          size={Platform.select({ android: 'large', ios: 'small' })}
          color={Colors.default.dark}
        />
      </Flex>
    </Flex>
  );
};

export default Loader;

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'rgba(1, 1, 1, 0.5)',
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1,
    position: 'absolute',
    top: 0,
    zIndex: 99,
    width: Dimensions.get('screen').width,
    height: Dimensions.get('screen').height,
  },
  loaderContainer: {
    width: scale(80),
    height: scale(80),
    borderRadius: mScale(16),
    backgroundColor: Colors.default.white,
    shadowColor: Colors.default.dark,
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.3,
    shadowRadius: 4.65,
    elevation: 8,
  },
});
