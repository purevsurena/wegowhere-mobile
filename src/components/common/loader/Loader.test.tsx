import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '@/components/common/loader/Loader';

test('Loader component renders correctly', () => {
  const tree = renderer.create(<Loader />).toJSON();
  expect(tree).toMatchSnapshot();
});
