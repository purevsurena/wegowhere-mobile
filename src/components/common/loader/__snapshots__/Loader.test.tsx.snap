// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "alignSelf": undefined,
      "backgroundColor": "rgba(1, 1, 1, 0.5)",
      "flex": 1,
      "flexDirection": undefined,
      "flexWrap": undefined,
      "height": 1334,
      "justifyContent": "center",
      "position": "absolute",
      "top": 0,
      "width": 750,
      "zIndex": 99,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": undefined,
        "backgroundColor": "#ffffff",
        "borderRadius": 24,
        "elevation": 8,
        "flex": undefined,
        "flexDirection": undefined,
        "flexWrap": undefined,
        "height": 160,
        "justifyContent": "center",
        "shadowColor": "#000000",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4.65,
        "width": 160,
      }
    }
  >
    <ActivityIndicator
      color="#000000"
      size="small"
    />
  </View>
</View>
`;
